sim/pokemon.ts

```diff
@@ -1590,11 +1589,11 @@ export class Pokemon {
 	}
 
 	eatItem(force?: boolean, source?: Pokemon, sourceEffect?: Effect) {
-		if (!this.item || this.itemState.knockedOff) return false;
+		if (!this.item) return false;
 		if ((!this.hp && this.item !== 'jabocaberry' && this.item !== 'rowapberry') || !this.isActive) return false;
-
 		if (!sourceEffect && this.battle.effect) sourceEffect = this.battle.effect;
 		if (!source && this.battle.event && this.battle.event.target) source = this.battle.event.target;
+
```

Comment:
I kept this in for readability, can remove it

Reply:
Nothing in this file should have changed, you even have a regression in here.

Reply:
i think the regression was for removing the knock off hardcode

Reply:
No, it's from https://github.com/smogon/pokemon-showdown/commit/0e692d4a9e5e11db9d650eb7a4ea84f99f43b071 and it's important.

Reply:
oh my bad.